client-rpc/src/rpc/sync_rpc.rs

```diff
@@ -5,9 +5,23 @@ use crate::server::to_rpc_error;
 use client_common::tendermint::Client;
 use client_common::Storage;
 use client_core::synchronizer::PollingSynchronizer;
+use client_core::wallet::syncer::ProgressReport;
 use client_core::wallet::syncer::{ObfuscationSyncerConfig, WalletSyncer};
 use client_core::wallet::WalletRequest;
 use client_core::TransactionObfuscation;
+use std::sync::Arc;
+use std::sync::Mutex;
+
+pub trait CBindingCallback: Send + Sync {
+    fn progress(&self, current: u64, start: u64, end: u64) -> i32;
+    fn set_user(&mut self, user: u64);
+    fn get_user(&self) -> u64;
```

Comment:
are these needed?

Reply:
because all calls are synchronous,
technically, not necessary,
but it's handy to give context info in callback.

Reply:
because all calls are synchronous,
technically, not necessary,
but it's handy to give context info in callback.

Reply:
I mean `set_user` and `get_user` aren't called anywhere?

Reply:
set_user is called in jsonrpc.rs of cro_clib,
get_user is not called in rust, can be removed

Reply:
can you point to the line where `set_user` is called?

Reply:
https://github.com/leejw51crypto/chain/blob/cro-1132g/cro-clib/src/jsonrpc.rs
line 328
it's another pr

Reply:
I see in this pr https://github.com/crypto-com/chain/pull/1196/files#diff-cb49bbc10a53700d54f20f45422fcc13R328